routers/base.go

```diff
@@ -91,17 +91,22 @@ func getUsernameByClientIdSecret(ctx *context.Context) (string, error) {
 	return fmt.Sprintf("app/%s", application.Name), nil
 }
 
-func getUsernameByKeys(ctx *context.Context) string {
+func getUsernameByKeys(ctx *context.Context) (string, error) {
 	accessKey, accessSecret := getKeys(ctx)
 	user, err := object.GetUserByAccessKey(accessKey)
 	if err != nil {
-		panic(err)
+		return "", err
+	}
+
+	if user == nil {
+		return "", fmt.Errorf("user not found for access key: %s", accessKey)
 	}
 
-	if user != nil && accessSecret == user.AccessSecret {
-		return user.GetId()
+	if accessSecret != user.AccessSecret {
```

Comment:
Why?

Reply:
Thought i would change it to align more with the `getUsernameByClientIdSecret` function above allowing explicit error handling.